python/cudf/cudf/_lib/groupby.pyx

```diff
@@ -133,19 +109,50 @@ cdef class GroupBy:
         from cudf.core.column_accessor import ColumnAccessor
         cdef vector[libcudf_groupby.aggregation_request] c_agg_requests
         cdef Column col
+        cdef Aggregation agg_obj
 
-        aggregations = _drop_unsupported_aggs(values, aggregations)
+        allow_empty = all(len(v) == 0 for v in aggregations.values())
 
+        included_aggregations = defaultdict(list)
+        idx = 0
         for i, (col_name, aggs) in enumerate(aggregations.items()):
             col = values._data[col_name]
+            dtype = col.dtype
+
+            valid_aggregations = (
+                _LIST_AGGS if is_list_dtype(dtype)
+                else _STRING_AGGS if is_string_dtype(dtype)
+                else _CATEGORICAL_AGGS if is_categorical_dtype(dtype)
+                else _STRING_AGGS if is_struct_dtype(dtype)
+                else _INTERVAL_AGGS if is_interval_dtype(dtype)
+                else _DECIMAL_AGGS if is_decimal_dtype(dtype)
+                else "ALL"
+            )
+            if (valid_aggregations is _DECIMAL_AGGS
+                    and rmm._cuda.gpu.runtimeGetVersion() < 11000):
+                raise RuntimeError(
+                    "Decimal aggregations are only supported on CUDA >= 11 "
+                    "due to an nvcc compiler bug."
+                )
+
             c_agg_requests.push_back(
                 move(libcudf_groupby.aggregation_request())
             )
-            c_agg_requests[i].values = col.view()
             for agg in aggs:
-                c_agg_requests[i].aggregations.push_back(
-                    move(make_aggregation(agg))
-                )
+                agg_obj = make_aggregation(agg)
+                if (valid_aggregations == "ALL"
+                        or agg_obj.kind in valid_aggregations):
+                    included_aggregations[col_name].append(agg)
+                    c_agg_requests[idx].aggregations.push_back(
+                        move(agg_obj.c_obj)
+                    )
+            if c_agg_requests[idx].aggregations.empty():
+                c_agg_requests.pop_back()
+            else:
+                c_agg_requests[idx].values = col.view()
+                idx += 1
+        if c_agg_requests.empty() and not allow_empty:
+            raise DataError("No numeric types to aggregate")
```

Comment:
is this the right error message? 
(non numeric types are also supported now and this error message is outdated)

Reply:
Yes, this is outdated. Perhaps something like this would be better:
> All requested aggregations are unsupported for column of the given dtypes.

Reply:
It looks like this error exists to match the pandas error verbatim. We test this in `test_groupby.py::test_raise_data_error`. While this error is underspecified based on `cudf` features, I'm not sure if we should change it or if we're OK with diverging from pandas at the level of the text of the exception.

Reply:
@kkraus14 @shwina do we need to match pandas verbatim here, or is it sufficient to throw the same exception type but have a different message? We support meaningful aggregations on non-numeric types, but this error alone could be misleading and suggest otherwise.

Reply:
I think it's sufficient to match the error type